components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -25,9 +36,50 @@ BraveWalletProviderImpl::BraveWalletProviderImpl(
 
 BraveWalletProviderImpl::~BraveWalletProviderImpl() {}
 
+bool BraveWalletProviderImpl::OnAddEthereumChainRequest(
+    const std::string& json_payload,
+    RequestCallback callback) {
+  if (!delegate_)
+    return false;
+  base::Value payload;
+  if (!ParsePayload(json_payload, "params", &payload))
+    return false;
+  std::vector<EthereumChain> chains = ValueToEthereumChain(payload);
+  if (chains.empty()) {
+    return false;
+  }
+  DCHECK_LT(chains.size(), size_t(2));
+  // TODO(spylogsster): Add support for multiple chains;
+  auto chain = chains[0];
+  if (GetNetworkURL(prefs_, chain.chain_id).is_valid()) {
+    OnChainAddedResult(std::move(callback), std::string());
+    return true;
+  }
+  // By https://eips.ethereum.org/EIPS/eip-3085 only chain id is required
+  // we expect chain name and rpc urls as well at this time
```

Comment:
cc @bbondy @yrliou

Reply:
Just post a follow up issue for this, thx.

Reply:
https://github.com/brave/brave-browser/issues/17637